### PR TITLE
rose.variable.array_split: fix left behind escape

### LINKED ISSUE #1363: rose metadata: remove backslash from escaped characters
Giving a setting the following metadata:
`values=foo\,bar, baz`

results in the literal `foo\,bar` being used instead of `foo,bar`.

This should be fixed.

### PR DESCRIPTION
**Pull request by matthewrmshin**
The main intention of this change is to ensure that escape characters
are removed after parsing the value.

The implementation now uses Python's standard `shlex` module, which
allows our logic to be simplified.

Fix #1363.

### CODE REVIEW

**Comment by benfitzpatrick** [comment]
(Test battery not happy)

**Comment by matthewrmshin** [comment]
@benfitzpatrick No, not ready yet.

**Comment by matthewrmshin** [comment]
I have just realised that the original logic is trying to preserve quote characters, hence it makes sense to preserve escape characters in the tokens as well. This means that the `shlex` solution will not work in the general case.

**Comment by matthewrmshin** [comment]
Test battery should be happy now.

**Comment by matthewrmshin** [comment]
@benfitzpatrick Please review. I think this is as good as it can get based on existing logic.

File: t/rose-macro/05-value-values-check.t [hunk 1]
```diff
@@ -203,4 +204,46 @@ file_cmp "$TEST_KEY.err" "$TEST_KEY.err" <<__CONTENT__
...
+values = 'Exeter\, England', 'Aberdeen\, Scotland'
+__META_CONFIG__
+run_pass "${TEST_KEY}" rose macro -V --config='../config'
+teardown
+
+TEST_KEY="${TEST_KEY_BASE}-comma-esc-bad"
+setup
+init_meta <<'__META_CONFIG__'
+[namelist:values_nl1=my_location]
+values = 'Manchester\, England', 'Glasgow\, Scotland'
```
Comment (benfitzpatrick) [inline t/rose-macro/05-value-values-check.t]: Could you test non-quoted values?

Comment (matthewrmshin) [inline t/rose-macro/05-value-values-check.t]: Non-quoted values are already covered by the `my_raw` setting in the original sets of tests.

**Comment by benfitzpatrick** [comment]
Shouldn't cause any problems with most of our real world metadata.

**Pull request merged**